satpy/readers/nwcsaf_nc.py

```diff
@@ -174,7 +175,9 @@ def scale_dataset(self, dsid, variable, info):
                                         coords=variable.coords, dims=variable.dims, attrs=variable.attrs)
 
             val, idx = np.unique(variable.attrs['palette_meanings'], return_index=True)
-            variable.attrs['palette_meanings'] = val
+            # The palette_meanings are applicaple to the unscaled values. Need to scale them:
+            # FIXME! Scale and offset of the palete_meanings are not available!
+            variable.attrs['palette_meanings'] = val * scale + offset
```

Comment:
Does this solve the problem, or is it not the right solution ?

Reply:
No, this doesn't solve it, tried that but realized it is NOT the scale/offset of the data, but the ancillary dataset `cmaprob_pal`

Reply:
ok, so does the palette itself have a scale and offset attribute that we discard ?

Reply:
Fixed in the last commits